matrix_sdk/src/room/common.rs

```diff
@@ -169,13 +170,30 @@ impl Common {
     }
 
     async fn ensure_members(&self) -> Result<()> {
+        if !self.are_events_visible() {
+            return Err(
+                error::Error::Http(HttpError::NotClientRequest)
```

Comment:
The whole point of this endeavor is to not make HTTP requests that we know will fail, we should just return `Ok(())` here since we ensured that we did everything we can to synchronize the members.

Reply:
Right, but I figured that we might want the information that the request was not made in the calling function. Throwing a recoverable error therefore seemed to be the way to go, since you can also just ignore the error if you don't need to handle it. 
Let me know if I should just return `Ok(())` instead.

Reply:
As you can see, that method is private. It's used to automatically synchronize the members, since that isn't possible unless we join or the history visibility changes there's no course of action the SDK can take to recover from this.

Reply:
I see. How about using a custom enum type?
In that case, we might return `Ok(RequestMade)` and `Ok(RequestNotMade)` so information is propagated without throwing an `Err`.
Again just throwing ideas, let me know if you'd rather just return `Ok(())` here

Reply:
Not really necessary since it won't be propagated to the users, you can add a log line mentioning what's going on, that's something users could introspect.

Reply:
Got it, changed it to return `Ok(())` instead